packages/modals/src/styled/StyledTooltipModalClose.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import styled, { css, DefaultTheme, ThemeProps } from 'styled-components';
+import { getColor, retrieveComponentStyles, DEFAULT_THEME } from '@zendeskgarden/react-theming';
+
+const COMPONENT_ID = 'modals.tooltip_modal.close';
+
+const colorStyles = (props: ThemeProps<DefaultTheme>) => {
+  const backgroundColor = 'primaryHue';
+  const foregroundColor = 'neutralHue';
+
+  return css`
+    background-color: transparent;
+    color: ${getColor(foregroundColor, 600, props.theme)};
+
+    &:hover {
+      background-color: ${getColor(backgroundColor, 600, props.theme, 0.08)};
+      color: ${getColor(foregroundColor, 700, props.theme)};
+    }
+
+    &[data-garden-focus-visible] {
+      box-shadow: ${props.theme.shadows.md(
+        getColor(backgroundColor, 600, props.theme, 0.35) as string
+      )};
+    }
+
+    &:active {
+      /* prettier-ignore */
+      transition:
+        background-color 0.1s ease-in-out,
+        color 0.1s ease-in-out;
+      background-color: ${getColor(backgroundColor, 600, props.theme, 0.2)};
+      color: ${getColor(foregroundColor, 800, props.theme)};
+    }
+  `;
+};
+
+/**
+ * 1. Remove dotted outline from Firefox on focus.
+ */
+export const StyledTooltipModalClose = styled.button.attrs({
```

Comment:
Is this a near copy/paste of https://github.com/zendeskgarden/react-components/blob/main/packages/modals/src/styled/StyledClose.ts? How might this be refactored to take advantage of extended, reduced, and common code?